src/lib/panorama.ts

```diff
@@ -0,0 +1,245 @@
+import * as THREE from 'three';
+
+import { fetch } from './utils';
+
+const TAU = 2 * Math.PI;
+const geometryCache = {};
+
+let oldPano: THREE.Group | THREE.Mesh | null = null;
+let newPano: THREE.Group | THREE.Mesh | null = null;
+let panoFadeTime = performance.now();
+
+export function animatePano(scene: THREE.Scene) {
+	scene.children
```

Comment:
This kind of comes out of no where:

```suggestion
	// FIX(Mikarific): remove residue pano meshes that may not have been removed by the animation not running.
	// I.e.: when the tab is out of focus
	scene.children
```

Reply:
Thinking about it better, maybe this could be a separate function altogether that runs on a timer? `.forEach`-ing over all children every frame is unnecessarily expensive

Reply:
I've fixed this in [the latest commit](https://github.com/netux/internet-roadtrip-LookOutTheWindow/pull/1/commits/7ce098e399766a08df68a37dbc392276dc45c084). Now instead of a `.filter` and a `.forEach` every frame, it does a single loop over the scene's children only if the tab is backgrounded and only on every panorama change.